src/catalog/src/system.rs

```diff
@@ -330,6 +331,7 @@ pub fn decode_system_catalog(
                 schema_name: table_parts[1].to_string(),
                 table_name: table_meta.table_name,
                 table_id,
+                engine: table_meta.engine.unwrap_or(MITO_ENGINE.to_string()),
```

Comment:
Can we not assume it's a mito table here? Not making engine field in `table_meta` optional.

Reply:
> Can we not assume it's a mito table here? Not making engine field in `table_meta` optional.

But if I remove the optional here, we still need to treat the empty string as the mito table to provide compatibility🥲